test/aave-v3/TestSupplyVault.t.sol

```diff
@@ -490,27 +490,141 @@ contract TestSupplyVault is TestSetupVaults {
         assertEq(token.balanceOf(_to), _toTransfer);
     }
 
-    function testAccrueRewardsToCorrectUser() public {
-        uint256 amount = 1e6 ether;
+    function testRewardsShouldAccrueWhenDepositingOnBehalf() public {
+        uint256 amount = 10_000 ether;
+        address[] memory poolTokens = new address[](1);
+        poolTokens[0] = aDai;
+
+        vaultSupplier1.depositVault(daiSupplyVault, amount);
+        vm.warp(block.timestamp + 10 days);
+
+        uint256 expectedTotalRewardsAmount = rewardsManager.getUserRewards(
+            poolTokens,
+            address(daiSupplyVault),
+            rewardToken
+        );
+
+        // Should update the unclaimed amount
+        vaultSupplier2.depositVault(daiSupplyVault, amount, address(vaultSupplier1));
+        (, uint128 userReward1) = daiSupplyVault.userRewards(rewardToken, address(vaultSupplier1));
+
+        vm.warp(block.timestamp + 10 days);
+
+        (uint128 index2, uint128 userReward2) = daiSupplyVault.userRewards(
+            rewardToken,
+            address(vaultSupplier2)
+        );
+        assertEq(index2, 0);
+        assertEq(userReward2, 0);
+        assertGt(uint256(userReward1), 0);
+        assertApproxEqAbs(uint256(userReward1), expectedTotalRewardsAmount, 10000);
+    }
 
-        vm.startPrank(address(supplier1));
-        ERC20(dai).approve(address(daiSupplyVault), type(uint256).max);
-        daiSupplyVault.deposit(amount, address(supplier1));
+    function testRewardsShouldAccrueWhenMintingOnBehalf() public {
+        uint256 amount = 10_000 ether;
+        address[] memory poolTokens = new address[](1);
+        poolTokens[0] = aDai;
 
+        vaultSupplier1.depositVault(daiSupplyVault, amount);
         vm.warp(block.timestamp + 10 days);
 
-        daiSupplyVault.redeem(
-            daiSupplyVault.balanceOf(address(supplier1)),
-            address(supplier2),
-            address(supplier1)
+        uint256 expectedTotalRewardsAmount = rewardsManager.getUserRewards(
+            poolTokens,
+            address(daiSupplyVault),
+            rewardToken
+        );
+        morpho.updateIndexes(aDai);
```

Comment:
this is unecessary as indexes should be updated when supplier2 mints shares of the dai vault right?
Not a blocker though

Reply:
I think it's to have the correct `previewMint`